i6_models/parts/conformer/frontend.py

```diff
@@ -0,0 +1,282 @@
+        if isinstance(self.pool_1_kernel_size, int):
+            assert self.pool_1_kernel_size % 2 == 1, "ConformerVGGFrontendV1 only supports odd kernel sizes"
+        if isinstance(self.pool_2_kernel_size, int):
+            assert self.pool_2_kernel_size % 2 == 1, "ConformerVGGFrontendV1 only supports odd kernel sizes"
+
+    def __post__init__(self):
+        super().__post_init__()
+        self.check_valid()
+
+
+class ConformerVGGFrontendV1(nn.Module):
+    """
+    Convolutional Front-End
+
+    Uses explicit padding for ONNX exportability, see:
+    https://github.com/pytorch/pytorch/issues/68880
+    """
+
+    def __init__(self, model_cfg: ConformerVGGFrontendV1Config):
+        """
+        :param model_cfg: model configuration for this module
+        """
+        super().__init__()
+
+        model_cfg.check_valid()
+
+        conv_padding = self._get_padding(model_cfg.conv_kernel_size)
+        pool_1_padding = self._get_padding(model_cfg.pool_1_kernel_size)
+        pool_2_padding = self._get_padding(model_cfg.pool_2_kernel_size)
+
+        self.conv1 = nn.Conv2d(
+            in_channels=model_cfg.features,
+            out_channels=model_cfg.channels_conv_1_2,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.conv2 = nn.Conv2d(
+            in_channels=model_cfg.channels_conv_1_2,
+            out_channels=model_cfg.channels_conv_1_2,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.pooling1 = nn.MaxPool2d(
+            kernel_size=model_cfg.pool_1_kernel_size,
+            stride=model_cfg.pool_1_strides,
+            padding=pool_1_padding,
+        )
+        self.conv3 = nn.Conv2d(
+            in_channels=model_cfg.channels_conv_1_2,
+            out_channels=model_cfg.channels_conv_3_4,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.conv4 = nn.Conv2d(
+            in_channels=model_cfg.channels_conv_3_4,
+            out_channels=model_cfg.channels_conv_3_4,
+            kernel_size=model_cfg.conv_kernel_size,
+            padding=conv_padding,
+        )
+        self.pooling2 = nn.MaxPool2d(
+            kernel_size=model_cfg.pool_2_kernel_size,
+            stride=model_cfg.pool_2_strides,
+            padding=pool_2_padding,
+        )
+        self.activation = model_cfg.activation
+
+    def forward(self, tensor: torch.Tensor) -> torch.Tensor:
+        """
+        T might be reduced to T' depending on ...
+
+        :param tensor: input tensor of shape [B,T,F]
+        :return: torch.Tensor of shape [B,T',F]
+        """
+        # conv 2d layers expect shape [B,F,T,C] so we have to transpose here
+        tensor = torch.transpose(tensor, 1, 2)  # [B,F,T]
+        # and add a dim
+        tensor = tensor[:, :, :, None]  # [B,F,T,C]
```

Comment:
The comment on the shape is wrong. You are not adding the channel dim here. The feature dim *is* the channel dim (the second dim always). You are adding another spatial dim here.
```suggestion
        tensor = tensor[:, :, :, None]  # [B,F,T,H]
```
But this makes me realize: I think you are using the VGG in the wrong way. Actually, I think usually the VGG operates on the time and feature dim, and for the first 2d-conv layer, it adds a dummy channel dim. So it means, above you have:
```python
self.conv1 = nn.Conv2d(1, conv1_channels, ...)
```
And the `transpose` above is not needed. So `tensor` is still [B,T,F]. And here you would do:
```suggestion
        tensor = tensor[:, None]  # [B,C=1,T,F]
```
In this comment, I keep "F" as it was the name of the original dim, but "F" here is now used as a spatial dim for the convolution.